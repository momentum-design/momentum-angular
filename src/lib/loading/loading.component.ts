import { Component, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'cui-loading, CSLoading',
  template: `
    <div class="cui-loading" #loader>
      <span class="cui-loading__icon"></span>
      <span class="cui-loading__icon"></span>
      <span class="cui-loading__icon"></span>
    </div>
  `,
  styles: [],
  encapsulation: ViewEncapsulation.None,
})
export class LoadingComponent {
  constructor() {}
}

 /**
 * @component loading
 * @section default
 * @angular
 *
<div className="row">
  <div className="docs-example docs-example--spacing" [style]="{fontSize: '1rem', display: 'flex'}">
    <cui-loading></cui-loading>
  </div>
  <div className="docs-example docs-example--spacing" [style]="{fontSize: '2rem', display: 'flex'}">
    <cui-loading></cui-loading>
  </div>
  <div className="docs-example docs-example--spacing" [style]="{fontSize: '3rem', display: 'flex'}">
    <cui-loading></cui-loading>
  </div>
</div>
 */
