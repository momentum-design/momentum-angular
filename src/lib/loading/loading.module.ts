import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoadingComponent } from './loading.component';

@NgModule({
  imports: [CommonModule],
  exports: [LoadingComponent],
  declarations: [LoadingComponent],
})
export class LoadingModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: LoadingModule,
      providers: [],
    };
  }
}
