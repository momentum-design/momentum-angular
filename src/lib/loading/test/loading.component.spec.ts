import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { LoadingComponent } from '../loading.component';

describe('LoadingComponent', () => {
  let component: LoadingComponent;
  let fixture: ComponentFixture<LoadingComponent>;
  let loadingNativeElement: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [LoadingComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LoadingComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render 3 span children', () => {
    loadingNativeElement = fixture.nativeElement;
    const spans = loadingNativeElement.querySelectorAll('span');

    expect(spans.length).toEqual(3);
    spans.forEach(span => {
      expect(span.className).toEqual('md-loading__icon');
    });
  });
});
