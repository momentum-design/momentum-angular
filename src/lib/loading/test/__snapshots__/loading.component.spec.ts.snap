// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingComponent should create and match snapshot 1`] = `
<cui-loading>
  <div
    class="cui-loading"
  >
    <span
      class="cui-loading__icon"
    />
    <span
      class="cui-loading__icon"
    />
    <span
      class="cui-loading__icon"
    />
  </div>
</cui-loading>
`;
